rajawali/src/main/java/c/org/rajawali3d/sceneview/SceneView.java

```diff
@@ -0,0 +1,227 @@
+package c.org.rajawali3d.sceneview;
+
+import android.graphics.Rect;
+import android.opengl.GLES20;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import c.org.rajawali3d.annotations.GLThread;
+import c.org.rajawali3d.annotations.RequiresReadLock;
+import c.org.rajawali3d.camera.Camera;
+import c.org.rajawali3d.engine.Engine;
+import c.org.rajawali3d.engine.RenderModel;
+import c.org.rajawali3d.engine.RenderView;
+import c.org.rajawali3d.object.renderers.ObjectRenderer;
+import c.org.rajawali3d.scene.Scene;
+import c.org.rajawali3d.scene.graph.NodeMember;
+import java.util.List;
+import java.util.concurrent.locks.Lock;
+import net.jcip.annotations.GuardedBy;
+import org.rajawali3d.math.Matrix4;
+import org.rajawali3d.util.RajLog;
+
+/**
+ * Author: Randy Picolet
+ */
+
+public class SceneView implements RenderView {
+    private static final String TAG = "SceneView";
+
+    @Nullable
+    private Engine engine;
+
+    @NonNull
+    private Scene scene;
+
+    // TODO probably need a decorator Viewport class around the Rect to flag changes...
+    @NonNull
+    private Rect viewport;
+
+    /**
+     * The {@link Camera} currently being used to render the scene.
+     */
+    @GuardedBy("nextCameraLock")
+    private Camera currentCamera;
+
+    private Camera nextCamera; // The camera the scene should switch to on the next frame.
+    private final Object nextCameraLock = new Object(); // Camera switching lock
+
+    @NonNull
+    @GLThread
+    private Matrix4 viewProjectionMatrix = new Matrix4();
+
+    @Nullable
+    @GLThread
+    private ObjectRenderer lastUsedRenderer; // Reference to the last used object engine
+
+    @Nullable
+    Lock currentlyHeldWriteLock;
+    @Nullable
+    Lock currentlyHeldReadLock;
+
+    // TODO Not sure about volatile semantics here
+    private volatile boolean isDisabled;
+
+    public SceneView(@NonNull Rect viewport) {
+        this.viewport = viewport;
+    }
+
+    public SceneView(@NonNull Scene scene) {
+        this.viewport = new Rect();
+        this.scene = scene;
+    }
+
+    public SceneView(@NonNull Rect viewport, @NonNull Scene scene) {
```

Comment:
With the class members `viewport` and `scene` being `@NonNull`, this should be the only constructor. Remove the other two. Another alternative is to set the value of `this.viewport` in the declaration and in the constructors do `this.viewport.set(viewport)` or whatever the actual syntax is.